lib/Db/PimcoreConfigStorage.php

```diff
@@ -0,0 +1,349 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Db;
+
+use Pimcore\Config;
+use Pimcore\Model\Tool\SettingsStore;
+use Sabre\VObject\Settings;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @internal
+ */
+final class PimcoreConfigStorage
```

Comment:
I'd move all this code directly into the `PimcoreConfigBagDao` , I don't think we need to put that into an extra class.

Reply:
PimcoreConfigBagListingDao for listing